controllers/clustergroupupgrade_controller.go

```diff
@@ -1012,12 +1030,64 @@ func (r *ClusterGroupUpgradeReconciler) buildRemediationPlan(ctx context.Context
 	}
 	r.Log.Info("Remediation plan", "remediatePlan", remediationPlan)
 	clusterGroupUpgrade.Status.RemediationPlan = remediationPlan
+
+	return nil
 }
 
 func getResourceName(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, policyName string) string {
 	return clusterGroupUpgrade.Name + "-" + policyName
 }
 
+func (r *ClusterGroupUpgradeReconciler) getAllClustersForUpgrade(ctx context.Context, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) ([]string, error) {
+	var clusterNames []string
+	keys := make(map[string]bool)
+	for _, clusterSelector := range clusterGroupUpgrade.Spec.ClusterSelector {
+		selectorList := strings.Split(clusterSelector, "=")
```

Comment:
Including a comment here with the format of the clusterSelector string would be helpful in understanding the logic.